unlock-app/src/components/interface/checkout/main/Select.tsx

```diff
@@ -177,10 +177,14 @@ export function Select({ checkoutService, injectedProvider }: Props) {
       const items = await Promise.all(
         Object.entries(paywallConfig.locks).map(async ([lock, props]) => {
           const networkId: number = props.network || paywallConfig.network || 1
-          const [lockData, fiatPricing] = await Promise.all([
-            web3Service.getLock(lock, networkId),
-            getFiatPricing(config, lock, networkId),
-          ])
+
+          const lockData = await web3Service.getLock(lock, networkId)
+          const fiatPricing = await getLockUsdPrice({
+            network: networkId,
+            currencyContractAddress: lockData?.currencyContractAddress,
+            amount: Number(lockData.keyPrice),
+          })
```

Comment:
Here need to do it sequentially because  `currencyContractAddress` and `keyPrice` are not available before